physical/oci/oci.go

```diff
@@ -0,0 +1,380 @@
+		permitPool:     physical.NewPermitPool(MaxNumberOfPermits),
+		namespaceName:  namespaceName,
+		haEnabled:      haEnabled,
+		lockBucketName: lockBucketName,
+	}, nil
+}
+
+func (o *Backend) Put(ctx context.Context, entry *physical.Entry) error {
+	o.logger.Debug("PUT started")
+	defer metrics.MeasureSince(metricPutFull, time.Now())
+	startAcquirePool := time.Now()
+	metrics.SetGauge(metricPermitsUsed, float32(o.permitPool.CurrentPermits()))
+	o.permitPool.Acquire()
+	defer o.permitPool.Release()
+	metrics.MeasureSince(metricPutAcquirePool, startAcquirePool)
+
+	defer metrics.MeasureSince(metricPut, time.Now())
+	size := int64(len(entry.Value))
+	opcClientRequestId, err := uuid.GenerateUUID()
+	if err != nil {
+		metrics.IncrCounter(metricPutFailed, 1)
+		o.logger.Error("failed to generate UUID")
+		return errwrap.Wrapf("failed to generate UUID: {{err}}", err)
+	}
+
+	o.logger.Debug("PUT", "opc-client-request-id", opcClientRequestId)
+	request := objectstorage.PutObjectRequest{
+		NamespaceName:      &o.namespaceName,
+		BucketName:         &o.bucketName,
+		ObjectName:         &entry.Key,
+		ContentLength:      &size,
+		PutObjectBody:      ioutil.NopCloser(bytes.NewReader(entry.Value)),
+		OpcMeta:            nil,
+		OpcClientRequestId: &opcClientRequestId,
+	}
+
+	resp, err := o.client.PutObject(ctx, request)
+	if resp.RawResponse != nil && resp.RawResponse.Body != nil {
+		defer resp.RawResponse.Body.Close()
+	}
+
+	if err != nil {
+		metrics.IncrCounter(metricPutFailed, 1)
+		return errwrap.Wrapf("failed to put data: {{err}}", err)
+	}
+
+	o.logRequest("PUT", resp.RawResponse, resp.OpcClientRequestId, resp.OpcRequestId, err)
+	o.logger.Debug("PUT completed")
+
+	return nil
+}
+
+func (o *Backend) Get(ctx context.Context, key string) (*physical.Entry, error) {
+	o.logger.Debug("GET started")
+	defer metrics.MeasureSince(metricGetFull, time.Now())
+	metrics.SetGauge(metricPermitsUsed, float32(o.permitPool.CurrentPermits()))
+	startAcquirePool := time.Now()
+	o.permitPool.Acquire()
+	defer o.permitPool.Release()
+	metrics.MeasureSince(metricGetAcquirePool, startAcquirePool)
+
+	defer metrics.MeasureSince(metricGet, time.Now())
+	opcClientRequestId, err := uuid.GenerateUUID()
+	if err != nil {
+		o.logger.Error("failed to generate UUID")
+		return nil, errwrap.Wrapf("failed to generate UUID: {{err}}", err)
+	}
+	o.logger.Debug("GET", "opc-client-request-id", opcClientRequestId)
+	request := objectstorage.GetObjectRequest{
+		NamespaceName:      &o.namespaceName,
+		BucketName:         &o.bucketName,
+		ObjectName:         &key,
+		OpcClientRequestId: &opcClientRequestId,
+	}
+
+	resp, err := o.client.GetObject(ctx, request)
+	if resp.RawResponse != nil && resp.RawResponse.Body != nil {
+		defer resp.RawResponse.Body.Close()
+	}
+	o.logRequest("GET", resp.RawResponse, resp.OpcClientRequestId, resp.OpcRequestId, err)
+
+	if err != nil {
+		if resp.RawResponse != nil && resp.RawResponse.StatusCode == http.StatusNotFound {
```

Comment:
@vuhpham-oracle I think there is something wrong with this check, because I'm seeing error in the logs for basic 404s:

```
2019-08-20T22:23:17.737Z [ERROR] storage.oci: GET client:opc-request-id   opc-request-id:   status-code: 404: error="Service error:ObjectNotFound. The object 'logical/0619b7bf-8285-b542-3d5f-c6c1c8669edd/b' was not found in the bucket 'XXXXXXX'. http status code: 404. Opc request id: iad-0:vzOiKzlDixeRWCfRH40/+mE1G6PQ7Ey5b8RMkyztRDzZklQwrXAzRXqrnc0ros0/"
```

Reply:
Nice catch, thanks. I updated to only error log with err !=nil and statusCode is 5xx.